app/org/sagebionetworks/bridge/services/UserProfileServiceImpl.java

```diff
@@ -1,27 +1,67 @@
 package org.sagebionetworks.bridge.services;
 
+import org.apache.commons.lang3.StringUtils;
+import org.sagebionetworks.bridge.crypto.AesGcmEncryptor;
 import org.sagebionetworks.bridge.models.User;
 import org.sagebionetworks.bridge.models.UserProfile;
 
 import com.stormpath.sdk.account.Account;
+import com.stormpath.sdk.directory.CustomData;
 
 public class UserProfileServiceImpl implements UserProfileService {
 
     private AuthenticationService authService;
     
+    private AesGcmEncryptor healthCodeEncryptor;
+
     public void setAuthenticationService(AuthenticationService authService) {
         this.authService = authService;
     }
+    
+    public void setHealthCodeEncryptor(AesGcmEncryptor encryptor) {
+        this.healthCodeEncryptor = encryptor;
+    }
 
     @Override
+    public UserProfile getProfile(String email) {
+        Account account = authService.getAccount(email);
+
+        UserProfile profile = new UserProfile();
+        profile.setFirstName(removeEmpty(account.getGivenName()));
+        profile.setLastName(removeEmpty(account.getSurname()));
+        profile.setUsername(removeEmpty(account.getUsername()));
+        profile.setEmail(removeEmpty(account.getEmail()));
```

Comment:
Are empty usernames and emails possible? We get the account from Stormpath using email, so I would assume email addresses always exist. Similarly, our signUp API provides a username, so I assume we'll always have a username.